bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/ThingHandlerKeypad.java

```diff
@@ -26,25 +38,84 @@
  */
 @NonNullByDefault
 public class ThingHandlerKeypad extends CaddxBaseThingHandler {
-    /**
-     * Constructor.
-     *
-     * @param thing
-     */
+    private final Logger logger = LoggerFactory.getLogger(ThingHandlerKeypad.class);
+
     public ThingHandlerKeypad(Thing thing) {
         super(thing, CaddxThingType.KEYPAD);
     }
 
     @Override
     public void updateChannel(ChannelUID channelUID, String data) {
+        if (channelUID.getId().equals(CaddxBindingConstants.KEYPAD_KEY_PRESSED)) {
+            StringType stringType = new StringType(data);
+            updateState(channelUID, stringType);
+        }
     }
 
     @Override
-    public void handleCommand(ChannelUID channelUID, Command command) {
+    public void caddxEventReceived(CaddxEvent event, Thing thing) {
+        logger.trace("caddxEventReceived(): Event Received - {}.", event);
+
+        if (getThing().equals(thing)) {
+            CaddxMessage message = event.getCaddxMessage();
+            CaddxMessageType mt = message.getCaddxMessageType();
+            ChannelUID channelUID = null;
+
+            // Log event messages have special handling
+            if (CaddxMessageType.KEYPAD_MESSAGE_RECEIVED.equals(mt)) {
+                for (CaddxProperty p : mt.properties) {
+                    if (!("".equals(p.getId()))) {
+                        String value = message.getPropertyById(p.getId());
+                        channelUID = new ChannelUID(getThing().getUID(), p.getId());
+                        updateChannel(channelUID, value);
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+        }
     }
 
     @Override
-    public void caddxEventReceived(CaddxEvent event, Thing thing) {
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
         updateStatus(ThingStatus.ONLINE);
```

Comment:
I think you forgot to remove this.

Reply:
I did intentionally not remove this.
The keypad has a function that it can act as an input device. To do that first the enterTerminalMode() action has to be executed and then the keypad will receive an event if a button is pressed.
If the keypad does not enter terminal mode, it will stay offline.

Reply:
But the next line just sets a new status anyway...

Reply:
You are right. I needed a coffee...
Corrected.